src/sphinxcontrib/towncrier/_towncrier.py

```diff
@@ -16,6 +16,12 @@
         load_config_from_file,
     )
 
+try:
+    from dataclasses import asdict, is_dataclass  # noqa: WPS433
+except ImportError:
+    # pylint: disable=unnecessary-lambda-assignment
```

Comment:
This many ignores indicates that something can be improved. Specifically, lambdas should only be used when passed as function arguments inline. Ignoring it is unacceptable and defeats the point of having linters. The proper solution is to follow the rule recommendation and declare a proper function definition.
Another solution could be assigning `None` and checking for `is None` in runtime.

Reply:
Mypy would fail with assigning None

Reply:
Not necessarily. This passes the whole suite (I didnt think about setting it to None):
```py
"""Towncrier related shims."""

from pathlib import Path
from typing import Any, Callable, Dict, Optional


try:
    # Towncrier >= 22.8.0rc1
    # pylint: disable=import-error,no-name-in-module
    from towncrier._settings.load import (  # noqa: WPS433, WPS436
        load_config_from_file,
    )
except ImportError:
    # pylint: disable=import-error,no-name-in-module
    from towncrier._settings import (  # noqa: WPS433, WPS436, WPS440
        load_config_from_file,
    )

is_dataclass: Optional[Callable[[Any], bool]]
try:
    from dataclasses import asdict, is_dataclass  # noqa: WPS433
except ImportError:
    is_dataclass = None  # noqa: WPS440


def get_towncrier_config(
        project_path: Path,
        final_config_path: Path,
) -> Dict[str, Any]:  # FIXME: add a better type  # pylint: disable=fixme
    """Return the towncrier config dictionary."""
    try:
        # Towncrier >= 19.9.0
        config = load_config_from_file(
            str(project_path), str(final_config_path),
        )
    except TypeError:
        # Towncrier < 19.9.0
        return load_config_from_file(str(final_config_path))
    if is_dataclass and is_dataclass(config):
        # Towncrier >= 22.12.0
        return asdict(config)
    return config
```